### PR TITLE
Removing FindROOT macro, add pythia8 config

### PR DESCRIPTION
**Pull request by cbernet**
This PR mainly fixes compilation issues on the mac, and also moves from the use FindRoot.cmake to CMake config. I think this requires root to be compiled with cmake, but this is more straightforward and modern. 

Other changes: 
* pythia 8 configuration files: ensuring random seed in Z->ddbar; added a config file. 
* pythia 8 generation: now printing cross-section

### CODE REVIEW

**Comment by cbernet** [comment]
hmm weird I can't reproduce this error. This error msg

```
[ 62%] Built target fccphysics-toolsDict-dictgen
Error: Cannot find fccphysics-toolsDict.cxx_tmp!
Error: Renaming fccphysics-toolsDict.cxx_tmp into fccphysics-toolsDict.cxx!
make[2]: *** [tools/fccphysics-toolsDict.cxx] Error 1
make[2]: *** Deleting file `tools/fccphysics-toolsDict.cxx'
make[1]: *** [tools/CMakeFiles/fccphysics-tools.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 62%] Built target fcc-physics-read
[ 62%] Built target fcc-physics-read-delphes
[ 68%] Linking CXX shared library libfccphysics-myanalysis.so
[ 75%] Built target fccphysics-myanalysis
[ 81%] Linking CXX executable fcc-pythia8-generate
[ 81%] Built target fcc-pythia8-generate
make: *** [all] Error 2
```

is what I was getting before this commit https://github.com/HEP-FCC/fcc-physics/pull/20/commits/995096f812231f430a18b8ac1cf42257ea88ef4e and now everything's ok on my side... Any idea on how I could investigate?

**Comment by jlingema** [comment]
Hi Colin, I think you missed the second dictionary generation done in `tools/` if you add the same line in there, it should be fine?

**Comment by cbernet** [comment]
retest this please

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DDROP_CGAL ${CPPSTDFLAGS} -Wall -
 
 # Make sure we find the Find*.cmake functions distributed with this package
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-set(CMAKE_PREFIX_PATH $ENV{PODIO} $ENV{FCCEDM})
+set(CMAKE_PREFIX_PATH $ENV{ROOTSYS} $ENV{PODIO} $ENV{FCCEDM})
```
Comment (jlingema) [inline CMakeLists.txt]: This entire line is not needed when using the fcc init script. May still be useful for compilation outside the lxplus environment.

The change should not be necessary, it's already added in [thisroot.sh](https://github.com/root-project/root/blob/master/config/thisroot.sh#L83)

**Review by jlingema — COMMENTED** [review]
Hi Colin,

I have one small comment, see below.

File: CMakeLists.txt (hunk 1 above)
Comment (cbernet) [reply CMakeLists.txt]: thank you, removed it.

**Comment by cbernet** [comment]
retest this please.

**Pull request merged**